detect.go

```diff
@@ -28,17 +29,23 @@ type ProjectParser interface {
 	NodeIsRequired(path string) (bool, error)
 }
 
-func Detect(buildpackYMLParser BuildpackConfigParser, configParser ConfigParser, projectParser ProjectParser) packit.DetectFunc {
+func Detect(buildpackYMLParser BuildpackConfigParser, configParser ConfigParser, projectParser ProjectParser, logger scribe.Logger) packit.DetectFunc {
```

Comment:
Is the logger in detect still necessary if we aren't logging the warning here anymore?